yarn-project/noir-libs/noir-aztec/src/types/vec.nr

```diff
@@ -37,18 +37,46 @@ impl<T, MaxLen> BoundedVec<T, MaxLen> {
         self.len -= 1;
         elem
     }
+
+    fn any<Env>(self, predicate: fn[Env](T) -> bool) -> bool {
+        let mut ret = false;
+        for i in 0..MaxLen {
+            if (i < self.len as u64) {
+                ret |= predicate(self.storage[i]);
+            }
+        }
+        ret
+    }
+}
+
+#[test]
+fn test_vec_push_pop() {
+    let mut vec: BoundedVec<Field, 3> = BoundedVec::new(0);
+    assert(vec.len == 0);
+    vec.push(2);
+    assert(vec.len == 1);
+    vec.push(4);
+    assert(vec.len == 2);
+    vec.push(6);
+    assert(vec.len == 3);
+    let x = vec.pop();
+    assert(vec.len == 2);
+    assert(x == 6);
+}
+
+#[test(should_fail)]
+fn test_vec_push_overflow() {
+    let mut vec: BoundedVec<Field, 1> = BoundedVec::new(0);
+    vec.push(1);
+    vec.push(2);
 }
 
-// #[test]
-// fn test_vec() {
-//     let vec: BoundedVec<Field, 2> = BoundedVec::new(0);
-//     assert(vec.len == 0);
-//     let vec1 = vec.push(1);
-//     assert(vec1.len == 1);
-//     let vec2 = vec1.push(1);
-//     assert(vec2.len == 2);
-//     let vec3 = vec2.push(1);
-//     assert(vec3.len == 3);
-//     let x = vec3.pop();
-//     assert(x == 1);
-// }
\ No newline at end of file
+#[test]
+fn test_vec_any() {
+    let mut vec: BoundedVec<Field, 3> = BoundedVec::new(0);
+    vec.push(0);
+    vec.push(1);
+    vec.push(2);
+    assert(vec.any(|v| v == 2) == true);
+    assert(vec.any(|v| v == 3) == false);
```

Comment:
Could we add a check to assert that it doesn't match values which are _greater_ than the `len`? E.g. push `1` to a new `vec`, and then `assert(vec.any(|v| v == 0) == false);` (i.e. check that it doesn't match `0` values that come after `len`.

Reply:
Good idea! Will do :)